carp.deployments.core/src/commonMain/kotlin/dk/cachet/carp/deployments/application/ParticipationService.kt

```diff
@@ -42,13 +42,13 @@ interface ParticipationService : ApplicationService<ParticipationService, Partic
     suspend fun getParticipantDataList( studyDeploymentIds: Set<UUID> ): List<ParticipantData>
 
     /**
-     * Set participant [data] for the given [inputDataType] in the study deployment with [studyDeploymentId].
+     * Set participant [data] in the study deployment with [studyDeploymentId].
      *
      * @throws IllegalArgumentException when:
      *   - there is no study deployment with [studyDeploymentId]
-     *   - [inputDataType] is not configured as expected participant data in the study protocol
-     *   - [data] is invalid data for [inputDataType]
+     *   - one or more of the keys in [data] isn't configured as expected participant data in the study protocol
+     *   - one or more of the set [data] isn't valid for the corresponding input data type
      * @return All data for the specified study deployment, including the newly set data.
      */
-    suspend fun setParticipantData( studyDeploymentId: UUID, inputDataType: InputDataType, data: Data? ): ParticipantData
+    suspend fun setParticipantData( studyDeploymentId: UUID, data: Map<InputDataType, Data?> ): ParticipantData
```

Comment:
Also I'm not quite sure why `data` is nullable. If something is optional, the caller should simply not set it unless there actually is a value to set.

Reply:
Good point. I did not do this purposefully. It was just a copy/paste. _But_, by making it nullable you can _unset_ values. Otherwise, this is not supported.

In other words, setting only a subset of fields is still possible. But passing null is to support removing previously set data.

Reply:
Documentation should be updated with the additional "unset" semantics then :)